ksql-engine/src/main/java/io/confluent/ksql/KsqlEngine.java

```diff
@@ -112,51 +112,47 @@
   private final QueryIdGenerator queryIdGenerator;
   private final KafkaClientSupplier clientSupplier;
 
-  public KsqlEngine(final KsqlConfig ksqlConfig) {
-    this(
-        new KafkaTopicClientImpl(ksqlConfig.getKsqlAdminClientConfigProps()),
-        new KsqlSchemaRegistryClientFactory(ksqlConfig).create(),
-        new DefaultKafkaClientSupplier(),
-        new MetaStoreImpl(new InternalFunctionRegistry())
-    );
-  }
+  private final KsqlConfig initializationKsqlConfig;
 
-  public KsqlEngine(final KafkaTopicClient kafkaTopicClient,
-                    final SchemaRegistryClient schemaRegistryClient,
-                    final KafkaClientSupplier clientSupplier
-  ) {
+  public KsqlEngine(final KsqlConfig initializationKsqlConfig) {
     this(
-        kafkaTopicClient,
-        schemaRegistryClient,
-        clientSupplier,
-        new MetaStoreImpl(new InternalFunctionRegistry())
+        new KafkaTopicClientImpl(initializationKsqlConfig.getKsqlAdminClientConfigProps()),
+        new KsqlSchemaRegistryClientFactory(initializationKsqlConfig).create(),
+        new DefaultKafkaClientSupplier(),
+        new MetaStoreImpl(new InternalFunctionRegistry()),
+        initializationKsqlConfig
     );
   }
 
   // called externally by tests only
   public KsqlEngine(final KafkaTopicClient topicClient,
                     final SchemaRegistryClient schemaRegistryClient,
-                    final MetaStore metaStore
+                    final MetaStore metaStore,
+                    final KsqlConfig initializationKsqlConfig
```

Comment:
It looks like the only reason to pass the `KsqlConfig` is to get the service id. Would it make sense to just pass the service id and store it directly? This would reduce the surface area of the dependencies.

Reply:
The main constructor receives KsqlConfig anyway and the other ones only are used for tests or the context class.